### PR TITLE
Upgrade to Realm Core 13.1.2

### LINKED ISSUE #5174: Upgrade to Realm Core v13.1.2

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #5174

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* ~[ ] 🚦 Tests~
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* ~[ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)~
* [ ] 📱 Check the React Native/other sample apps work if necessary
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~

### CODE REVIEW

File: src/js_subscriptions.hpp [hunk 1]
```diff
@@ -139,13 +139,13 @@ template <typename T>
 void SubscriptionClass<T>::get_name(ContextType ctx, ObjectType this_object, ReturnValue& return_value)
 {
     auto sub = get_internal<T, SubscriptionClass<T>>(ctx, this_object);
-    auto name = sub->name();
+    auto name = sub->name;
```
Comment (RedBeard0531) [inline src/js_subscriptions.hpp:142]:
```suggestion
    const auto& name = sub->name;
```

No need to force an extra copy. Or just use `sub->name` directly everywhere.

File: src/js_subscriptions.hpp [hunk 2]
```diff
@@ -139,13 +139,13 @@ template <typename T>
...
 
-    if (name == "") {
-        return_value.set_null();
+    if (name) {
+        return_value.set(name);
```
Comment (RedBeard0531) [inline src/js_subscriptions.hpp:145]:
```suggestion
        return_value.set(*name);
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,9 +13,8 @@
 * File format: generates Realms with format v23 (reads and upgrades file format v5 or later for non-synced Realm, upgrades file format v10 or later for synced Realms).
 
 ### Internal
-<!-- * Either mention core version or upgrade -->
-<!-- * Using Realm Core vX.Y.Z -->
-<!-- * Upgraded Realm Core from vX.Y.Z to vA.B.C -->
+* Upgraded Realm Core from v13.1.1 to v13.1.2. ([#5174](https://github.com/realm/realm-js/issues/5174))
+* Migrated to the new `Subscription` API in Realm Core.
```
Comment (RedBeard0531) [inline CHANGELOG.md:17]: This has no real impact on users and was a small code change so I don't know if it is worth calling out in release notes. Your call.

**Review by RedBeard0531 — APPROVED** [review]
Glad to see that it simplified the consumption patterns where expected.

File: CHANGELOG.md (hunk 3 above)
Comment (kneth) [reply CHANGELOG.md:17]: I'll remove it.

**Review by takameyer — APPROVED** [review]
LGTM 🚢

**Pull request merged**